### ISSUE TITLE
Assets::generateFileVersion() does not work with Bedrock installations

### ISSUE THREAD

**Issue by sybbear** [htmlburger/wpemerge-theme#40]
## Version
  - WP Emerge Starter Theme: 0.15.1
  - WP Emerge (if not the default):
  - WordPress / Bedrock: 5.3.2
  - PHP: 7.2

## Expected behavior
`wpemerge-theme-core/src/Assets/Assets::generateFileVersion()` should return file versions with Bedrock Wordpress installations.

## Actual behavior
The function always returns false, even that asset file exists.

## Steps to reproduce (in case of a bug)
  1. Install Wordpress using bedrock
  1. Create a WP Emerge theme
  1. On page load, that theme assets are not versioned in the URL

## Comments
We are using [Bedrock boilerplate](https://roots.io/bedrock/) for wordpress projects, as it provides modern file structure, supports managing plugins, themes and localizations through composer and  treats Wordpress as a dependency.
Bedrock has the following project structure by default:
```
- project
   - web (webroot, /)
       - app
          - mu-plugins
          - plugins
          - themes
          - languages
       - wp (Wordpress folder)
```
The issue is that while generally WP Emerge works fine with Bedrock, the function `wpemerge-theme-core/src/Assets/Assets::generateFileVersion()`  does not return a file version.

## How to resolve (example `project.com`)
- In normal wordpress installations home_url and site_url are the same. `https://project.com`
- In Bedrock, home_url will be `https://project.com` and site_url will be `https://project.com/wp`

1. https://github.com/htmlburger/wpemerge-theme-core/blob/0.15.0/src/Assets/Assets.php#L56
The line should use `home_url` instead of `site_url`
2. https://github.com/htmlburger/wpemerge-theme-core/blob/0.15.0/src/Assets/Assets.php#L63 There should be an extendable way to replace home_url with project root path. 
The fix for our project was to remove `wp/` from ABSPATH:
```php
			// Generate the absolute path to the file
			$file_path = str_replace(
				[$home_url, '/'],
				[str_replace('wp/', '', ABSPATH), DIRECTORY_SEPARATOR],
				$src
			);
```

## Bottom line
Could you implement a fix to this or suggest a way to override `generageFileVersion` ?

**Comment by atanas-dev** [comment]
Thank you for the detailed write-up @vikingmaster !

Due to the holidays I'm not exactly sure when I can release a fix (hopefully this weekend), but in the mean time you can use one of two work-arounds:
1. Edit assets.php
2. Replace `enqueueScript()` and `enqueueStyle()` usage with the equivalent from here:
https://github.com/htmlburger/wpemerge-theme-core/blob/master/src/Assets/Assets.php#L110-L134
3. Use hardcoded version numbers / filemtime() with semi-hardcoded file paths.

OR

1. Copy-paste the offending function and apply a fix manually.
2. Replace the usage in assets.php with your new function.

**Comment by sybbear** [comment]
Thank you for the hint :)
Looking forward to see the fix in upcoming updates, WP Emerge rocks! 🤘

**Comment by sybbear** [comment]
Hey @atanas-dev 
Wanted to check, is there a fix planned any time soon?

**Comment by atanas-dev** [comment]
@vikingmaster just released `0.15.1` of `htmlburger/wpemerge-theme-core` which fixes the issue.
To get the update use this:
`composer update htmlburger/wpemerge-theme-core`

**Comment by sybbear** [comment]
Very much appreciated!
